### PR TITLE
impr: multi-state event refactor part 2 (FPLA-1840)

### PR DESCRIPTION
**Pull request by joez-f**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1840

### Change description ###

Continuation of the multi-state refactor

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-definition/CaseEvent/MultiState.json [hunk 1]
```diff
@@ -483,5 +483,67 @@
...
+    "ShowEventNotes": "N",
+    "EndButtonLabel": "Save and continue",
+    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/manage-representatives/about-to-start",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/manage-representatives/about-to-submit",
+    "CallBackURLSubmittedEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/callback/manage-representatives/submitted"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "ID": "manageRepresentativesGatekeeping",
```
Comment (tomaszpowroznik) [inline ccd-definition/CaseEvent/MultiState.json:536]: why it is different, is it not a bug that submitted is not there. If is is special for gatekeeping then it is not multisate

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent.json [hunk 2]
```diff
@@ -62,6 +62,13 @@
     "UserRole": "caseworker-publiclaw-solicitor",
     "CRUD": "R"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseEventID": "enterChildren",
+    "UserRole": "caseworker-publiclaw-courtadmin",
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent.json:69]: what about gatekeeper, prev he had CRU

Comment (tomaszpowroznik) [reply ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent.json:69]: same for amendOtherProceedings, now gatekeeper lost CRU

File: ccd-definition/CaseEvent/MultiState.json (hunk 1 above)
Comment (joez-f) [reply ccd-definition/CaseEvent/MultiState.json:536]: it shouldn't have been there, my mistake 👍

File: ccd-definition/AuthorisationCaseEvent/DEPRECATED.json [hunk 3]
```diff
@@ -335,5 +335,19 @@
     "CaseEventID": "uploadC2Gatekeeping",
     "UserRole": "[SOLICITOR]",
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseEventID": "amendChildren",
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseEvent/DEPRECATED.json:342]: what about events like amendChildrenGatekeeping? I think gatekeeper should be able to see old events which are deprecated now

File: ccd-definition/AuthorisationCaseEvent/DEPRECATED.json [hunk 4]
```diff
@@ -335,5 +335,19 @@
...
+    "UserRole": "caseworker-publiclaw-courtadmin",
+    "CRUD": "CRU"
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseEvent/DEPRECATED.json:344]: should be R for deprecated event. It should be not possible to create deprecated event

File: ccd-definition/AuthorisationCaseEvent/DEPRECATED.json (hunk 3 above)
Comment (joez-f) [reply ccd-definition/AuthorisationCaseEvent/DEPRECATED.json:342]: you are right, will add + check other permissions that may be missing

**Pull request merged**